clients/clients-integration-tests/src/test/java/org/apache/kafka/clients/consumer/PlaintextConsumerCommitTest.java

```diff
@@ -0,0 +1,680 @@
+            assertEquals(1, cb.successCount);
+
+            // Try with coordinator known
+            consumer.commitAsync(Map.of(tp, new OffsetAndMetadata(2L)), cb);
+            consumer.commitSync(Map.of(tp1, new OffsetAndMetadata(2L)));
+
+            assertEquals(2, consumer.committed(Set.of(tp)).get(tp).offset());
+            assertEquals(2, consumer.committed(Set.of(tp1)).get(tp1).offset());
+            assertEquals(2, cb.successCount);
+
+            // Try with empty sync commit
+            consumer.commitAsync(Map.of(tp, new OffsetAndMetadata(3L)), cb);
+            consumer.commitSync(Map.of());
+
+            assertEquals(3, consumer.committed(Set.of(tp)).get(tp).offset());
+            assertEquals(2, consumer.committed(Set.of(tp1)).get(tp1).offset());
+            assertEquals(3, cb.successCount);
+        }
+    }
+
+    private Consumer<byte[], byte[]> createConsumer(GroupProtocol protocol, boolean enableAutoCommit) {
+        return cluster.consumer(Map.of(
+            GROUP_ID_CONFIG, "test-group",
+            GROUP_PROTOCOL_CONFIG, protocol.name().toLowerCase(Locale.ROOT),
+            ENABLE_AUTO_COMMIT_CONFIG, enableAutoCommit
+        ));
+    }
+    
+    private void sendRecords(int numRecords) {
+        try (Producer<byte[], byte[]> producer = cluster.producer()) {
+            sendRecords(producer, numRecords, tp, System.currentTimeMillis());
+        }
+    }
+
+    private void sendRecords(
+        Producer<byte[], byte[]> producer,
+        int numRecords,
+        TopicPartition topicPartition
+    ) {
+        sendRecords(producer, numRecords, topicPartition, System.currentTimeMillis());
+    }
+
+    private void sendRecords(
+        Producer<byte[], byte[]> producer, 
+        int numRecords, 
+        TopicPartition topicPartition,
+        long startingTimestamp
+    ) {
+        for (var i = 0; i < numRecords; i++) {
+            var timestamp = startingTimestamp + i;
+            var record = new ProducerRecord<>(
+                topicPartition.topic(),
+                topicPartition.partition(),
+                timestamp,
+                ("key " + i).getBytes(),
+                ("value " + i).getBytes()
+            );
+            producer.send(record);
+        }
+        producer.flush();
+    }
+
+    private void awaitAssignment(
+        Consumer<byte[], byte[]> consumer,
+        Set<TopicPartition> expectedAssignment
+    ) throws InterruptedException {
+        TestUtils.waitForCondition(() -> {
+            consumer.poll(Duration.ofMillis(100));
+            return consumer.assignment().equals(expectedAssignment);
+        }, "Timed out while awaiting expected assignment " + expectedAssignment + ". " +
+            "The current assignment is " + consumer.assignment()
+        );
+    }
+
+    private void sendAndAwaitAsyncCommit(
+        Consumer<byte[], byte[]> consumer,
+        Optional<Map<TopicPartition, OffsetAndMetadata>> offsetsOpt
```

Comment:
It looks like all `offsetOpt` input is not empty. Do you think that we can remove `Optional<>` from `sendAndAwaitAsyncCommit` and `RetryCommitCallback`?